ceph-releases/luminous/ubuntu/16.04/daemon/zap_device.sh

```diff
@@ -1,80 +1,155 @@
 #!/bin/bash
 set -e
 
+# Get a list of child devices from a root device
+function get_child_partitions {
+# $1: raw parent device
+  if [ ! -e  "${1}" ] && [ ! -b "${1}" ]; then
+    echo "Error, ${1} doesn't seem to be a valid device!"
+    exit 1
+  fi
+  PARTS=$(lsblk -no KNAME "${1}")
+  for p in $PARTS; do
+    kname=$(lsblk --nodeps -no KNAME /dev/"${p}")
+    pkname=$(lsblk --nodeps -no PKNAME /dev/"${p}")
+    if [ "${kname}" != "${pkname}" ] && [ -n "${pkname}" ]; then
+      echo /dev/"${p}"
+    fi
+  done
+}
+
+# Get parent device of a partition
+function get_parent_device {
+# $1: partition to get the parent's device (eg. /dev/sda2)
+  if [ ! -e "${1}" ] && [ ! -b "${1}" ]; then
+    echo "Error, ${1} doesn't seem to be a valid partition!"
+    exit 1
+  fi
+  lsblk --nodeps -no PKNAME "${1}"
+}
+
+function is_blockdevice {
+  if test -b "${1}"; then echo 1; else echo 0; fi
+}
+
+function is_parentpartition {
+  local result
+  result=$(lsblk --nodeps -no PKNAME "${1}")
+  if [ "$result" == "" ]; then
+    echo 0
```

Comment:
echo?
I guess you meant `return`, right?

Reply:
ditto